transport/http/src/test/java/org/kaazing/gateway/transport/http/connector/specification/rfc7230/ConnectionManagementIT.java

```diff
@@ -0,0 +1,450 @@
+                        closed2.countDown();
+                        return null;
+                    }
+                });
+            }
+        });
+        connector.connect("http://localhost:8080/", handler, new ConnectSessionInitializerGet());
+        closed2.await(2, SECONDS);
+
+        final CountDownLatch closed3 = new CountDownLatch(1);
+
+        context.checking(new Expectations() {
+            {
+                oneOf(handler).sessionCreated(with(any(IoSessionEx.class)));
+                oneOf(handler).sessionOpened(with(any(IoSessionEx.class)));
+                oneOf(handler).sessionClosed(with(any(IoSessionEx.class)));
+                will(new CustomAction("Latch countdown") {
+                    @Override
+                    public Object invoke(Invocation invocation) throws Throwable {
+                        closed3.countDown();
+                        return null;
+                    }
+                });
+            }
+        });
+        connector.connect("http://localhost:8080/", handler, new ConnectSessionInitializerGet());
+        closed3.await(2, SECONDS);
+
+        k3po.finish();
+    }
+
+    /*
+     * Server related tests - pipelining not supported client side.
+     * 
+     * @Specification({"server.should.accept.http.pipelining/response"}) public void serverShouldAcceptHttpPipelining()
+     * throws Exception {
+     */
+
+    /*
+     * Server related tests - pipelining not supported client side.
+     * 
+     * @Specification({"client.with.pipelining.must.not.retry.pipelining.immediately.after.failure/response"}) public
+     * void clientWithPipeliningMustNotRetryPipeliningImmediatelyAfterFailure() throws Exception {
+     */
+
+    @Test
+    @Specification({"server.must.close.its.half.of.connection.after.sending.response.if.it.receives.a.close/response"})
+    public void serverMustCloseItsHalfOfConnectionAfterSendingResponseIfItReceivesAClose() throws Exception {
+        final IoHandler handler = context.mock(IoHandler.class);
+        final CountDownLatch closed = new CountDownLatch(1);
+
+        context.checking(new Expectations() {
+            {
+                oneOf(handler).sessionCreated(with(any(IoSessionEx.class)));
+                oneOf(handler).sessionOpened(with(any(IoSessionEx.class)));
+                oneOf(handler).sessionClosed(with(any(IoSessionEx.class)));
+                will(new CustomAction("Latch countdown") {
+                    @Override
+                    public Object invoke(Invocation invocation) throws Throwable {
+                        closed.countDown();
+                        return null;
+                    }
+                });
+            }
+        });
+
+        connector.connect("http://localhost:8080/path", handler, new ConnectSessionInitializerGetConnection());
+        assertTrue(closed.await(2, SECONDS));
+
+        k3po.finish();
+    }
+
+    @Ignore("Doesn't Accept a second connection.")
```

Comment:
Can you try and fix this test? Maybe the k3po spec test origin server needs to be changed for the second `accept`. If it is a bug we can leave it ignored with the link to the bug.